app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -31,12 +35,28 @@ public void onCreatePreferences(final Bundle savedInstanceState, final String ro
 
         addPreferencesFromResourceRegistry();
 
-        initialSelectedLocalization = org.schabi.newpipe.util.Localization
-                .getPreferredLocalization(requireContext());
-        initialSelectedContentCountry = org.schabi.newpipe.util.Localization
-                .getPreferredContentCountry(requireContext());
         initialLanguage = defaultPreferences.getString(getString(R.string.app_language_key), "en");
 
+        if (Build.VERSION.SDK_INT >= 33) {
+            requirePreference(R.string.app_language_key).setVisible(false);
+            final Preference newAppLanguagePref =
+                    requirePreference(R.string.app_language_android_13_and_up_key);
+            newAppLanguagePref.setSummaryProvider(preference -> {
+                final Locale customLocale = AppCompatDelegate.getApplicationLocales().get(0);
+                if (customLocale != null) {
+                    return customLocale.getDisplayName();
+                }
+                return getString(R.string.systems_language);
+            });
+            newAppLanguagePref.setOnPreferenceClickListener(preference -> {
+                final Intent intent = new Intent(Settings.ACTION_APP_LOCALE_SETTINGS)
+                        .setData(Uri.fromParts("package", requireContext().getPackageName(), null));
+                startActivity(intent);
```

Comment:
That's a very edge case, but if the user has disabled settings or if some ROMs are broken (hello weird Android TV devices), this would crash the app due to no app being able to handle the intent.